analytical_engine/core/fragment/dynamic_fragment_view.h

```diff
@@ -0,0 +1,263 @@
+    return fragment_->GetLocalOutDegree(v);
+  }
+
+  inline int GetLocalInDegree(const vertex_t& v) const {
+    return fragment_->GetLocalInDegree(v);
+  }
+
+  inline bool Gid2Vertex(const vid_t& gid, vertex_t& v) const {
+    return fragment_->Gid2Vertex(gid, v);
+  }
+
+  inline vid_t Vertex2Gid(const vertex_t& v) const {
+    return fragment_->Vertex2Gid(v);
+  }
+
+  inline vid_t GetInnerVerticesNum() const {
+    return fragment_->GetInnerVerticesNum();
+  }
+
+  inline vid_t GetOuterVerticesNum() const {
+    return fragment_->GetOuterVerticesNum();
+  }
+
+  inline bool IsInnerVertex(const vertex_t& v) const {
+    return fragment_->IsInnerVertex(v);
+  }
+
+  inline bool IsOuterVertex(const vertex_t& v) const {
+    return fragment_->IsOuterVertex(v);
+  }
+
+  inline bool GetInnerVertex(const oid_t& oid, vertex_t& v) const {
+    return fragment_->GetInnerVertex(oid, v);
+  }
+
+  inline bool GetOuterVertex(const oid_t& oid, vertex_t& v) const {
+    return fragment_->GetOuterVertex(oid, v);
+  }
+
+  inline oid_t GetInnerVertexId(const vertex_t& v) const {
+    return fragment_->GetInnerVertexId(v);
+  }
+
+  inline oid_t GetOuterVertexId(const vertex_t& v) const {
+    return fragment_->GetOuterVertexId(v);
+  }
+
+  inline oid_t Gid2Oid(const vid_t& gid) const {
+    return fragment_->Gid2Oid(gid);
+  }
+
+  inline bool Oid2Gid(const oid_t& oid, vid_t& gid) const {
+    return fragment_->Oid2Gid(oid, gid);
+  }
+
+  inline bool InnerVertexGid2Vertex(const vid_t& gid, vertex_t& v) const {
+    return fragment_->InnerVertexGid2Vertex(gid, v);
+  }
+
+  inline bool OuterVertexGid2Vertex(const vid_t& gid, vertex_t& v) const {
+    return fragment_->OuterVertexGid2Vertex(gid, v);
+  }
+
+  inline vid_t GetOuterVertexGid(const vertex_t& v) const {
+    return fragment_->GetOuterVertexGid(v);
+  }
+
+  inline vid_t GetInnerVertexGid(const vertex_t& v) const {
+    return fragment_->GetInnerVertexGid(v);
+  }
+
+  inline bool IsAliveVertex(const vertex_t& v) const {
+    return fragment_->IsAliveVertex(v);
+  }
+
+  inline bool IsAliveInnerVertex(const vertex_t& v) const {
+    return fragment_->IsAliveInnerVertex(v);
+  }
+
+  inline bool IsAliveOuterVertex(const vertex_t& v) const {
+    return fragment_->IsAliveOuterVertex(v);
+  }
+
+  inline grape::DestList IEDests(const vertex_t& v) const {
+    return fragment_->IEDests(v);
+  }
+
+  inline grape::DestList OEDests(const vertex_t& v) const {
+    return fragment_->OEDests(v);
+  }
+
+  inline grape::DestList IOEDests(const vertex_t& v) const {
+    return fragment_->IOEDests(v);
+  }
+
+  inline adj_list_t GetOutgoingAdjList(const vertex_t& v) {
+    if (view_type_ == "reversed") {
+      return fragment_->GetIncomingAdjList(v);
+    }
+    return fragment_->GetOutgoingAdjList(v);
+  }
+
+  inline adj_list_t GetIncomingAdjList(const vertex_t& v) {
+    if (view_type_ == "reversed" || view_type_ == "directed") {
+      return fragment_->GetOutgoingAdjList(v);
```

Comment:
please double check:
if directed, then return out_going_adj for incoming request?

Reply:
the original graph of directed view is undirected graph,  its' edges all store in out_going_adj